appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/AssetList.java

```diff
@@ -95,15 +103,38 @@ public void onMouseMove(MouseMoveEvent event) {
         clientY = event.getClientY();
       }
     });
+    
+    
     assetList.addSelectionHandler(new SelectionHandler<TreeItem>() {
+      private int clickCount = 0;
       @Override
-      public void onSelection(SelectionEvent<TreeItem> event) {
-        TreeItem selected = event.getSelectedItem();
-        ProjectNode node = (ProjectNode) selected.getUserObject();
-        // The actual menu is determined by what is registered for the filenode
-        // type in CommandRegistry.java
-        ProjectNodeContextMenu.show(node, selected.getWidget(), clientX, clientY);
-      }});
+      public void onSelection(final SelectionEvent<TreeItem> event) {
+         clickCount++;
+         if (clickCount == 1) {
+            Timer timer = new Timer() {
+            @Override
+            public void run() {
+                if (clickCount == 1) {
+                   TreeItem selected = event.getSelectedItem();
+                   ProjectNode node = (ProjectNode) selected.getUserObject();
+                   // The actual menu is determined by what is registered for the filenode
+                   // type in CommandRegistry.java
+                   ProjectNodeContextMenu.show(node, selected.getWidget(), clientX, clientY);
+                }
+                if (clickCount == 2) {
+                   // to preview asset
+                   TreeItem selected = event.getSelectedItem();
+                   ProjectNode node = (ProjectNode) selected.getUserObject();
+                   PreviewFileCommand cmd = new PreviewFileCommand();
+                   cmd.execute(node);
+                } 
+                clickCount = 0;
```

Comment:
To ensure this always happens, I recommend that you place the code above in a `try {}` block and then have this reset in a `finally {}` block. That way if for whatever reason an exception is thrown above, the count will still get reset. Otherwise, the functionality would break until the page is reloaded.